### PR TITLE
[Provider/Azure] cache: Add mutex locks for public methods to avoid data race

### PR DESCRIPTION
**Pull request by gaurav1086**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Add mutex.Lock for set and delete for concurrent operations
**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
N/A
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by gaurav1086** [comment]
cc @andyzhangx

File: staging/src/k8s.io/legacy-cloud-providers/azure/cache/azure_cache.go [hunk 1]
```diff
@@ -152,6 +152,8 @@ func (t *TimedCache) Get(key string, crt AzureCacheReadType) (interface{}, error
 
 // Delete removes an item from the cache.
 func (t *TimedCache) Delete(key string) error {
+	t.Lock.Lock()
+	defer t.Lock.Unlock()
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/cache/azure_cache.go:156]: this is not needed, as there're already locks within t.Lock

**Comment by gaurav1086** [comment]
Closing as not needed.

**Pull request closed without merging**